### ISSUE TITLE
.NET framework 2.0 and earlier are not supported

### ISSUE THREAD

**Issue by hsluoyz** [ua-parser/uap-csharp#46]
First thanks for the work! We are planning to use uap-csharp in [Microsoft Cosmos's SCOPE](http://www.vldb.org/pvldb/1/1454166.pdf), which has an early version .NET framework support (maybe 2.0 or even earlier). I found this code has used delegate like ``Func`` return type, which is not supported by .NET framework 2.0 (see: https://docs.microsoft.com/en-us/dotnet/api/system.func-1?view=netframework-4.8#applies-to).

I don't know if such usage is necessary, however, I saw the [uap-cpp (C++ version)](https://github.com/ua-parser/uap-cpp) and it doesn't use any such features. Can we refactor the code to remove those advanced features to make it support .NET framework 2.0 and earlier? Thanks.

**Comment by hsluoyz** [comment]
@enemaerke @atifaziz

**Comment by atifaziz** [comment]
I think this is doable for .NET Framework 2.0 without any refactoring work. Any version earlier than that (1.x) isn't something I'd have time or motivation for.

**Comment by withinfocus** [comment]
Upvote for additional target support. Even if we could get a Framework 4.x slice in place things would be better off. Can't just target .NET Standard (especially just 1.x) as it brings in an enormous amount of portability dependencies.

**Comment by atifaziz** [comment]
I have a version targeting .NET 2.0 in my fork's [net20](https://github.com/atifaziz/uap-csharp/tree/net20) branch. I was going to submit it as a PR but the first [3](https://github.com/ua-parser/uap-csharp/pull/47) [simple](https://github.com/ua-parser/uap-csharp/pull/48) [ones](https://github.com/ua-parser/uap-csharp/pull/49) I have open haven't yet been merged so I'm guessing @enemaerke isn't finding the cycles.

**Comment by enemaerke** [comment]
@atifaziz I have just merged 2 of the 3 PRs this evening, the last one is showing some merge conflicts. I'll try to take a look at your net20 PR once you submit it

**Comment by atifaziz** [comment]
> @atifaziz I have just merged 2 of the 3 PRs this evening, the last one is showing some merge conflicts. I'll try to take a look at your net20 PR once you submit it

Thanks, man! I've resolve all conflicts with the last PR so as soon as that's in, I can submit a PR with the `net20` branch too.

**Comment by yuanrui** [comment]
hope support .net 3.5 and .net 4.0

**Comment by withinfocus** [comment]
@atifaziz I know it's been a bit but do you need any assistance with that next PR? We're eager to see Core support for our uses too.

**Comment by atifaziz** [comment]
@yuanrui & @withinfocus: While I have made several contributions to the project, the primary maintainer and owner is @enemaerke. He has permissions to merge and publish new releases so the questions should really be directed to him; you see, [I am still waiting for him to publish a version](https://github.com/ua-parser/uap-csharp/pull/50#issuecomment-512138509) with work from #50 merged (that pertains to this issue).

**Comment by withinfocus** [comment]
Yeah I'm just referring to your May 29 comment where it seemed you were waiting for all prior PRs to clear and merge which has now happened, irrelevant to a release being issued. I would think the Core work could go out with what's already done as one release.

**Comment by enemaerke** [comment]
Hi all and sorry for the long wait on this. As @atifaziz mentions, I'm struggling a bit to find some spare time to address the issues and PRs, so the work I'm doing is very scattered at current time.

However, I have published version 3.1.40 to NuGet a few minutes ago, that should include support for .Net2.0 and .NetCore2.0. I'm hoping that'll address the things brought up here, otherwise let me know
